config/manager/manager.yaml

```diff
@@ -60,5 +60,26 @@ spec:
           requests:
             cpu: 10m
             memory: 64Mi
+        volumeMounts:
+          - mountPath: /etc/webhook-cert
+            name: webhook-cert
+            readOnly: true
       serviceAccountName: gateway-api-controller
       terminationGracePeriodSeconds: 10
+      volumes:
+        - name: webhook-cert
+          secret:
+            defaultMode: 420
+            secretName: webhook-cert
+---
```

Comment:
This placeholder certificate and key were generated with:

```
openssl req -x509 -nodes -days -365 -newkey rsa:2048 -keyout ${KEY_FILE} -out ${CERT_FILE} -subj "/CN=not-a-real-cn/O=not-a-real-o"
```

When calling a webhook, the API server performs the following validations:
1. Cert validity window has not passed
2. SAN matches DNS (e.g. ```webhook-service.aws-application-networking-system.svc```)
3. Cert is signed by a known authority OR matches the ```ca``` field on the client configuration for the webhook (either signer or direct match)

Having a placeholder here is important because the controller's webhook server needs this value to start up successfully. The webhook server does not perform strict validation.

Once the secret is patched with a valid cert, Kubernetes will distribute the value and the webhook server will be notified when the files are updated on the volume mount. See [controller-runtime/pkg/webhook/server.go](https://github.com/kubernetes-sigs/controller-runtime/blob/main/pkg/webhook/server.go#L207) for more details.

Reply:
I've removed the placeholder values entirely and instead populate them during ```make build-deploy```